### PR TITLE
JS-259 Do not warn against active Node.js versions

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**

### CODE REVIEW

**Comment by yassin-kammoun-sonarsource** [comment]
The ticket mentions testing the CI against all active Node.js versions.
This will really slow down our pull requests.
Is it really something we want to do?

**Comment by yassin-kammoun-sonarsource** [comment]
The following dates have expired for a while now:

https://github.com/SonarSource/SonarJS/blob/5f03925e341be154fbbcbe22a607c9a2c4be78ac/sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/NodeDeprecationWarning.java#L43-L46

However, we still log the following message:

https://github.com/SonarSource/SonarJS/blob/5f03925e341be154fbbcbe22a607c9a2c4be78ac/sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/NodeDeprecationWarning.java#L65-L70

Should we officially acknowledge and announce that we don't support Node.16 and Node.17?

**Comment by saberduck** [comment]
@yassin-kammoun-sonarsource I think the decision about Node 16 was properly announced, so you can cleanup the code and remove all mentions of it https://community.sonarsource.com/t/node-js-16-is-no-longer-supported-in-the-analysis-environment/112167

File: sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/NodeDeprecationWarning.java [hunk 1]
```diff
@@ -54,8 +53,7 @@ public class NodeDeprecationWarning {
    */
   static final Version MIN_SUPPORTED_NODE_VERSION = Version.create(18, 17, 0);
   static final int MIN_RECOMMENDED_NODE_VERSION = 18;
-  static final List<String> RECOMMENDED_NODE_VERSIONS = List.of("^18.18.0", "^20.9.0");
-  static final List<Integer> ALL_RECOMMENDED_NODE_VERSIONS = Arrays.asList(18, 20, 21);
+  static final List<String> RECOMMENDED_NODE_VERSIONS = List.of("^18.18.0", "^20.9.0", "^22.9.0");
```
[the remark this answers is not in the document]
Comment (saberduck) [reply sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/NodeDeprecationWarning.java:56]: I believe the minor versions follow ESLint `package.json` `engine` field https://github.com/eslint/eslint/blob/main/package.json#L204 , I wouldn't update unless we have a reason to

**Comment by saberduck** [comment]
> The ticket mentions testing the CI against all active Node.js versions. This will really slow down our pull requests. Is it really something we want to do?

The tests would run in parallel, so they shouldn't slow down that much. However, it will consume more resources. I think testing against 18, 20, and 22 is reasonable.

File: sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/NodeDeprecationWarning.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yassin-kammoun-sonarsource) [reply sonar-plugin/bridge/src/main/java/org/sonar/plugins/javascript/bridge/NodeDeprecationWarning.java:56]: Alright, I will follow what's in there and replace `^22.9.0` I just added with `>=21.1.0` instead.

**Comment by yassin-kammoun-sonarsource** [comment]
> > The ticket mentions testing the CI against all active Node.js versions. This will really slow down our pull requests. Is it really something we want to do?
> 
> The tests would run in parallel, so they shouldn't slow down that much. However, it will consume more resources. I think testing against 18, 20, and 22 is reasonable.

Alright. I will tackle the CI changes in a separate pull request.

**Pull request merged**